pygbe/main.py

```diff
@@ -355,14 +367,27 @@ def main(argv=sys.argv, log_output=True, return_output_fname=False,
     print('-'*30)
     print('Total setup time   : {}s'.format(setup_time))
 
-    tic = time.time()
+
+    #   Check if there is a complex dielectric
+    complexDiel = 0
+    for f in field_array:
+        if numpy.iscomplexobj(f.E):
+            complexDiel = 1
+
 
     ### Solve
+    tic = time.time()
+
     print('Solve')
-    phi = numpy.zeros(param.Neq)
+    # Initializing phi dtype according to the problem we are solving.
+    if complexDiel == 1:
+        phi = numpy.zeros(param.Neq, type(f.E))
```

Comment:
Oh I see your point, yes it might be a good idea setting a variable for that as you said. Probably better to discuss what is the best option, since you mention you want to re-factor more.